### PR TITLE
Fix systests races, etc

### PR DESCRIPTION
**Pull request by patrickxb**
Found some races and other typos in systests

@mlsteele can you look at test_logger.go?
@maxtaco see env.go
@mmaxim see reachability

### CODE REVIEW

File: go/systests/team_invite_test.go [hunk 1]
```diff
@@ -366,7 +366,7 @@ func TestImpTeamWithMultipleRooters(t *testing.T) {
 			if (arg.TeamID.Eq(team1) || arg.TeamID.Eq(team2)) && arg.Changes.MembershipChanged && !arg.Changes.KeyRotated && !arg.Changes.Renamed && arg.LatestSeqno == toSeqno {
 				t.Logf("change matched with %q", arg.TeamID)
 				winningTeam = arg.TeamID
-				return
+				break
```
Comment (patrickxb) [inline go/systests/team_invite_test.go]: This test was busted...It doesn't pass now.

Comment (patrickxb) [reply go/systests/team_invite_test.go]: @zapu can you take a look?

File: go/logger/test_logger.go [hunk 2]
```diff
@@ -48,11 +49,13 @@ var _ Logger = (*TestLogger)(nil)
 
 // ctx can be `nil`
 func (log *TestLogger) common(ctx context.Context, lvl logging.Level, useFatal bool, fmts string, arg ...interface{}) {
+	log.Lock()
+	defer log.Unlock()
 	if log.log.Failed() {
-		failReported := atomic.SwapInt32(&(log.failReported), 1)
-		if failReported == 0 {
+		if !log.failReported {
```
Comment (mlsteele) [inline go/logger/test_logger.go]: log.Lock could be inside this block

**Comment by mlsteele** [comment]
test_logger.go looks good!

File: go/logger/test_logger.go (hunk 2 above)
Comment (patrickxb) [reply go/logger/test_logger.go]: I had it there originally, but I think there's a problem in systests with these "TEST FAILED" messages.  They appear multiple times.  I thought moving the lock up might fix it, but it didn't

Comment (mlsteele) [reply go/logger/test_logger.go]: That they appear multiple times is a shortcoming of the way it's done.

**Comment by mmaxim** [comment]
👍 on reachability, not sure how I missed that one.

File: go/logger/test_logger.go (hunk 2 above)
Comment (patrickxb) [reply go/logger/test_logger.go]: ok, thanks.

File: go/systests/team_invite_test.go (hunk 1 above)
Comment (zapu) [reply go/systests/team_invite_test.go]: woah, thanks for finding this. can you skip if it doesn't pass?

Comment (patrickxb) [reply go/systests/team_invite_test.go]: Yeah, i'll skip it for now in this branch.

Comment (patrickxb) [reply go/systests/team_invite_test.go]: i added a ticket for it

**Pull request merged**